static/app/stores/configStore/configProvider.tsx

```diff
@@ -1,54 +1,26 @@
-import {useEffect, useReducer, useRef} from 'react';
+import React, {useReducer, useState} from 'react';
 
 import LegacyConfigStore from 'sentry/stores/configStore';
 import {ConfigContext} from 'sentry/stores/configStore/configContext';
-import {
-  configReducer,
-  makeBridgableReducer,
-} from 'sentry/stores/configStore/configReducer';
+import {configReducer} from 'sentry/stores/configStore/configReducer';
 import {Config} from 'sentry/types';
 
 interface ConfigProviderProps {
   children: React.ReactNode;
   initialValue: Config;
-  bridgeReflux?: boolean;
 }
 
 export function ConfigProvider(props: ConfigProviderProps) {
-  const initialBridgeRefluxValue = useRef<ConfigProviderProps['bridgeReflux']>(
-    props.bridgeReflux
-  );
-  const contextValue = useReducer(
-    makeBridgableReducer(configReducer, props.bridgeReflux ?? false),
-    props.initialValue
-  );
-
-  useEffect(() => {
-    if (initialBridgeRefluxValue.current !== props.bridgeReflux) {
-      throw new Error(
-        `bridgeReflux must not change between rerenders. This may result in undefined and out of sync behavior between the two stores. bridgeReflux changed from ${initialBridgeRefluxValue.current} -> ${props.bridgeReflux}`
-      );
-    }
-
-    if (!props.bridgeReflux) {
-      return undefined;
-    }
-
-    const [_, dispatch] = contextValue;
-
-    // @ts-ignore we need to bind the listener to a context, hence the variable name,
-    // but sadly typescript complains about shadowing here
-    const bindContext = this;
-    const unsubscribeListener = LegacyConfigStore.listen(changes => {
-      dispatch({type: 'patch', payload: changes});
-    }, bindContext);
+  const [signalState, setSignalState] = useState({});
+  const contextValue = useReducer(configReducer, props.initialValue);
 
-    return () => {
-      unsubscribeListener();
-    };
-  }, [props.bridgeReflux]);
+  const dispatch = () => {
+    setSignalState({});
+  };
```

Comment:
@k-fish if I understood correctly, this is where you would proxy to the store correct? How would you go around implementing the dispatch signature here? If I understood your proposal correctly, we'd want to proxy to the Store here. I guess we could have something like

```tsx
const dispatch = (updateFn: ((store) => void)) => {
   updateFn(store)
   setSignalState({})
}
```

Reply:
So this works and it doesnt require a timeout, which is nice. Something I do find awkward here is that this always feels like the migration is half done. On one side, we are using the react context, but on the other side, we are still dispatching to the store (and we are doing it through the value we get from react context)... 

Once we would start migrating this, we'd basically need to extend updateFn to support action updates right? So something in the line of
```tsx
(updateFn: ((store) => void) | action) => {
  if(isAction()){
    if(isSetThemeAction){
      legacyConfigStore.setTheme(action.theme)
      setLocalThemeState(action.theme)
    } else if(isOtherAction){
      ...other action handlers
    }
    setSignalState({})
  }

   updateFn(store)
   setSignalState({})
}
```

Reply:
It seems that this problem boils down to what is the nicest way to keep both "stores" in sync. I see the upside of this approach in the ability to avoid timeouts, but I do not like the updateFn implementation and the fact that we are not dispatching to the react state and that we'll eventually have to come back to update all the actions too